msgraph-sdk-android-auth/src/main/java/com/microsoft/graph/authentication/Callback.java

```diff
@@ -0,0 +1,59 @@
+package com.microsoft.graph.authentication;
+
+import android.app.Activity;
+import android.app.Application;
+import android.os.Bundle;
+import android.util.Log;
+
+public class Callback implements Application.ActivityLifecycleCallbacks {
+
+    private Activity currentActivity;
+
+    private void setCurrentActivity(Activity activity) {
+        this.currentActivity = activity;
+    }
+
+    public Callback(Activity activity) {
+        setCurrentActivity(activity);
+    }
+
+    public Activity getActivity() {
+        return currentActivity;
+    }
+
+    @Override
+    public void onActivityCreated(Activity activity, Bundle savedInstanceState) {
+        setCurrentActivity(activity);
+    }
+
+    @Override
+    public void onActivityStarted(Activity activity) {
+        setCurrentActivity(activity);
+    }
+
+    @Override
+    public void onActivityResumed(Activity activity) {
+        setCurrentActivity(activity);
+    }
+
+    @Override
+    public void onActivityPaused(Activity activity) {
+        setCurrentActivity(activity);
+    }
+
+    @Override
+    public void onActivityStopped(Activity activity) {
+        setCurrentActivity(activity);
+    }
+
+    @Override
+    public void onActivitySaveInstanceState(Activity activity, Bundle outState) {
+        setCurrentActivity(activity);
+    }
+
+    @Override
```

Comment:
pls see if you are required to set current activity when current activity is getting stopped, paused or destroyed.

Reply:
I have resolved this comment now.